marimo/_save/cache.py

```diff
@@ -179,15 +200,38 @@ def update(
                     f"({type(value)}:{ref})."
                 )
 
+        # Convert objects to stubs in both defs and meta
         for key, value in self.defs.items():
-            # If it's a module we must replace with a stub.
-            if inspect.ismodule(value):
-                self.defs[key] = ModuleStub(value)
-            elif inspect.isfunction(value):
-                self.defs[key] = FunctionStub(value)
-            elif isinstance(value, UIElement):
-                # UIElement cannot be restored directly, so we store a stub.
-                self.defs[key] = UIElementStub(value)
+            self.defs[key] = self._convert_to_stub_if_needed(value)
+
+        if "return" in self.meta:
```

Comment:
I think this is fair to run on all meta keys. See https://github.com/marimo-team/marimo/pull/5935 where I was tinkering with the API

Sorry I didn't get this in at the time, I hadn't settled on the best way to expose "stubs" Let me know if you have feedback!